src/main/java/org/jvnet/hudson/test/PropertiesTestSuite.java

```diff
@@ -66,11 +78,43 @@ public synchronized Object put(Object key, Object value) {
                     return null;
                 }
             };
+
+            byte[] contents = IOUtils.toByteArray(resource.openStream());
+            if (!isEncoded(contents, StandardCharsets.US_ASCII)) {
+                boolean isUtf8 = isEncoded(contents, StandardCharsets.UTF_8);
+                boolean isIso88591 = isEncoded(contents, StandardCharsets.ISO_8859_1);
+                if (isUtf8 && isIso88591) {
+                    throw new AssertionError(resource + " is encoded with both valid utf-8 and ISO-8859-1 characters, the utf-8 characters need to be encoded with something like `native2ascii`");
```

Comment:
The message seems a bit confusing to me as there is really no such thing as a "UTF-8 character" or a "ISO-8859-1" character. In Java all strings are Unicode. I would rephrase this as follows (suggestion only, I know this is a bit verbose):

```suggestion
                    throw new AssertionError(resource + " is valid UTF-8 and valid ISO-8859-1. To avoid problems when auto-detecting the encoding, use the lowest common denominator of ASCII encoding and express non-ASCII characters with escape sequences using a tool like `native2ascii`.");
```

Feel free to change this if you want to make it shorter.